src/zonetree/answer.rs

```diff
@@ -247,8 +274,13 @@ pub struct AnswerAuthority {
     /// The NS record set if it should be included.
     ns: Option<SharedRrset>,
 
-    /// The DS record set if it should be included..
+    /// The DS record set if it should be included.
     ds: Option<SharedRrset>,
+
+    /// Any additional address records to include.
```

Comment:
This is conceptionally weird: The type represents the authority section, but these records go into the additional section.

Reply:
On re-reading this: It might be good to split out additional records into its own field in `Answer` and have a type `AnswerAdditional` that collects all the additional records? We need to distinguish between necessary and optional additional RRsets (the latter can be dropped if we lack space in the response).

Reply:
Done. Please check.